src/rantanplan/structures.py

```diff
@@ -11,224 +10,145 @@
+        return False
 
+    if not repeating_structure and len(lengths_list) != len(ranges_list):
+        return False
 
-def is_seguidilla_gitana(lengths):
-    """
-    [6, 6, 11, 6]
-    :param lengths:
-    :return:
-    """
-    correct_lines = 0
-    number_of_verses = len(lengths)
-    if number_of_verses == 4:
-        for i in (0, 1, 3):
-            if 8 > lengths[i] > 4:
-                correct_lines += 1
-        if 13 > lengths[2] > 9:
-            correct_lines += 1
-    return correct_lines == number_of_verses
+    lengths_list = itertools.cycle(lengths_list)
 
+    if allow_fluctuation:
+        return all(
+            any(r in verse_range for r in range(verse_length - fluctuation,
+                                                verse_length + fluctuation + 1))
+            for
+            verse_length, verse_range in zip(lengths_list, ranges_list))
 
-def is_estrofa_safica(lengths):
-    """
-    [11, 11, 11, 5]
-    :param lengths:
-    :return:
-    """
-    correct_lines = 0
-    number_of_verses = len(lengths)
-    if number_of_verses == 4:
-        for i in (0, 1, 2):
-            if 13 > lengths[i] > 8:
-                correct_lines += 1
-        if 7 > lengths[3] > 3:
-            correct_lines += 1
-    return correct_lines == number_of_verses
+    return all(
+        map(lambda verse_length, verse_range: verse_length in verse_range,
+            lengths_list, ranges_list))
 
 
-def is_estrofa_safica_unamuno(lengths):
+def has_same_length_verses(fixed_length, ranges_list):
     """
-    [11, 11, 7, 5]
-    :param lengths:
-    :return:
+    Checks id all ranges contain the same fixed value
+    :param fixed_length: ixed value to be checked
+    :param ranges_list: List of verse length ranges
+    :return: `True` if all verses pass the condition, `False` otherwise
     """
-    correct_lines = 0
-    number_of_verses = len(lengths)
-    if number_of_verses == 4:
-        for i in (0, 1):
-            if 13 > lengths[i] > 8:
-                correct_lines += 1
-        if 9 > lengths[2] > 5:
-            correct_lines += 1
-        if 7 > lengths[3] > 3:
-            correct_lines += 1
-    return correct_lines == number_of_verses
+    return all(fixed_length in length_range for length_range in ranges_list)
 
 
-def is_estrofa_francisco_de_la_torre(lengths):
+def has_mixed_length_verses(length_a, length_b, ranges_list):
     """
-    [11, 11, 11, 7]
-    :param lengths:
-    :return:
+    Given two numbers, checks if all ranges contains both of them, and only those numbers, at least once.
+    This function generates two binary numbers, one for each length, with `1` and `0` when that number is found within
+    a given position in teh range or not, respectively.
+    After that a logical `OR` is applied between the two numbers and the output is compared with binary of all `1` with
+    the length of the original list of ranges.
```

Comment:
Still pretty confusing.
```suggestion
    Given two numbers, checks whether all ranges contain both of them,
    and only those numbers, at least once.
    This function generates two binary numbers, one for each length,
    with `1` indicating the number is found within a given position in the
    range, and `0` when not found.
    After that, a logical `OR` is applied between the two numbers and the
    output is compared with a binary number with just `1`'s and the length of
    the original list of ranges.
```